sanitizers/src/main/java/com/code_intelligence/jazzer/sanitizers/XpathInjection.kt

```diff
@@ -0,0 +1,95 @@
+// Copyright 2022 Code Intelligence GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.code_intelligence.jazzer.sanitizers
+
+import com.code_intelligence.jazzer.api.FuzzerSecurityIssueHigh
+import com.code_intelligence.jazzer.api.HookType
+import com.code_intelligence.jazzer.api.Jazzer
+import com.code_intelligence.jazzer.api.MethodHook
+import com.code_intelligence.jazzer.api.MethodHooks
+import java.lang.invoke.MethodHandle
+
+/**
+ * Detects Xpath injections.
+ *
+ * Untrusted input has to be escaped in such a way that queries remain valid otherwise an injection
+ * could be possible. This sanitizer guides the fuzzer to inject insecure characters. If an exception
```

Comment:
I'm not familiar with Xpaths - could you explain why an injection would be problematic? What would be the possible impact?

Reply:
Xpath is used to query XML. Kinda similar to SQL Injection. If quotes in the user input are not sanitized they can be used to inject into the Xpath expression. This allows for example to crawl the XML file or skip password checks.
Examples here:
https://owasp.org/www-community/attacks/XPATH_Injection
https://owasp.org/www-community/attacks/Blind_XPath_Injection